### PR TITLE
Fixes HiveLord broodling spawn rate

### PR DESCRIPTION
**Pull request by EchoFade**
Fixes Undocumented bug:

Hivelords attack the player and spawn a hivelord brood. For whatever reason this happens seemingly once every server tic. This will quickly fill a tunnel with 10+ broodlings while the hivelord sits in the corner and laughs at you.

This code denotes how often the hivelord can spawn a brood, modify the spawn(30) timer to change broodling spawn time. Currently set to spawn one every 3 attacks. Current spawn rate gives max of 3 broodlings per actively attacking hivelord.

This also allows tweaks to the broodling spawn rate if that's your kinda thing.

ALSO,
because EATING THROUGH MASKS is such a huge deal for our dear coderbus; I've changed the wording of using hivelord cores for healing to make it so that it seems like you smear it on your suit instead of eating it. If this angers you still I feel I must remind you of spees magic and for you some chill pills

HG helped me get started on this and LordBistian helped me get this working. We tested on a fresh pull of the /tg/ master repository and it seems to work really well.

Apologies if I tried to merge this incorrectly.

Also credit to Man_Shroom for moral support. OLE!

### CODE REVIEW

**Comment by SconesC** [comment]
The number of commits is a reference to the absurd amount of broods spawned

**Comment by EchoFade** [comment]
@SconesC Yep that was totally intentional and totally not because I've never done this before and am blundering my way through it. Totally.

**Comment by AlexanderUlanH** [comment]
If we're going to increase the brood spawn rate, we should probably buff the broods in some way or another.  It seems like this change would move hivelords from "not a threat" to "free healing".

**Comment by WJohn** [comment]
Honestly I feel like hivelord cores really shouldn't lose their effect after a single minute. The ONLY THING you get out of them is healing yourself after fighting the hivelord, essentially it's as if it never existed in the first place.

**Comment by EchoFade** [comment]
@AlexanderUlanH Did you mean decrease the brood spawnrate? As I tried to say, the spawn rate is tweakable, I feel that 3 - 4 broods attacking you at once was fair to try to take it out versus the 8-10 that stay alive from the continual spawning.

I have a working version of the code setup on a personally hosted server. If you want we can schedule a time where you can test it out.

@WJohn 
most of the time i've encountered hivelords paired with another hivelord or paired with a goliath. The creatures spawning together or fairly near one another is not a rare occurrence. Usually you have to run from that mess or try to draw one of them away to ensure you don't get overwhelmed. 

This thought process is based off of single mining, if you have a partner with you that changes things. but most mining mobs are easy to kill with two people, and i feel that i see many people mining on their own and calling for backup if things start going poorly.

With your concern about the cores becoming useless to quickly I understand and I hear you.

Quick question due to my limited personal knowldge of byond code:

>/obj/item/asteroid/hivelord_core/New()
> 	spawn(1200)
> 		inert = 1

This ^ is the specific section that denotes how long the core is valid for. I think that 1200 is two minutes right? (I do not know for sure i am asking for confirmation)

I would be happy to tweak that as part of this bugfix but what really would be a fair amount of time till the core is useless? The way this works is with the H.revive() command which in situations outside of mining would potentially be unbalanced. 

There are ways I can think of to fix it instead of increasing the timer, however the point of this specific fix is to stop hivelord broods from spawning every single attack.

The eating hivelord core through the mask nonsense was brought up in coderbus while i was asking for help limiting broodling spawn and i threw in the re-wording as a stop-gap to make sure other people wouldn't poke at mining code.

**Comment by EchoFade** [comment]
Sorry about the close/open I clicked the wrong button!

File: code/modules/mob/living/simple_animal/hostile/mining_mobs.dm [hunk 1]
```diff
@@ -280,10 +286,10 @@
 				user << "<span class='notice'>[src] are useless on the dead.</span>"
 				return
 			if(H != user)
-				H.visible_message("[user] forces [H] to eat [src]... they quickly regenerate all injuries!")
+				H.visible_message("[user] forces [H] to apply [src]... they quickly regenerate all injuries!")
 			else
-				user << "<span class='notice'>You chomp into [src], barely managing to hold it down, but feel amazingly refreshed in mere moments.</span>"
-			playsound(src.loc,'sound/items/eatfood.ogg', rand(10,50), 1)
+				user << "<span class='notice'>You start to smear [src] on yourself. It feels and smells disgusting as it seeps into your suit, but you feel amazingly refreshed in mere moments.</span>"
```
Comment (Cheridan) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: >It feels and smells disgusting as it seeps into your suit

What if I'm naked and not wearing a suit? :^)

File: code/modules/mob/living/simple_animal/hostile/mining_mobs.dm [hunk 2]
```diff
@@ -241,16 +241,22 @@
...
 	pass_flags = PASSTABLE
-
+	var/recentlyspawned = 0
+// This code denotes how often the hivelord can spawn a brood, modify the below spawn(30) timer to change spawn time. currently set to every 3 seconds. Current spawn rate gives max of 3 broodlings per actively attacking hivelord - EchoFade
 /mob/living/simple_animal/hostile/asteroid/hivelord/OpenFire(var/the_target)
-	var/mob/living/simple_animal/hostile/asteroid/hivelordbrood/A = new /mob/living/simple_animal/hostile/asteroid/hivelordbrood(src.loc)
-	A.GiveTarget(target)
-	A.friends = friends
-	A.faction = faction
+	if(!recentlyspawned)
```
Comment (Jordie0608) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: You should use world.time for cooldowns instead of a var and spawn()

**Comment by phil235** [comment]
@EchoFade Are you still working on this ?

**Comment by EchoFade** [comment]
@phil235 yessir, Sorry about the delays. I finished finals last week and have been really apathetic to doing anything because I really wanted a break from thinking in any capacity.

Let me get back onto fixing the problem this weekend and implement/test the above suggested changes. 

Also I'll try to figure out how to revert the 12 commit thing. 

Again sorry for being inactive, just desperately needed a break.

File: code/modules/mob/living/simple_animal/hostile/mining_mobs.dm [hunk 3]
```diff
@@ -241,16 +241,22 @@
...
 	minimum_distance = 3
 	pass_flags = PASSTABLE
-
+	var/recentlyspawned = 0
+// This code denotes how often the hivelord can spawn a brood, modify the below spawn(30) timer to change spawn time. currently set to every 3 seconds. Current spawn rate gives max of 3 broodlings per actively attacking hivelord - EchoFade
 /mob/living/simple_animal/hostile/asteroid/hivelord/OpenFire(var/the_target)
-	var/mob/living/simple_animal/hostile/asteroid/hivelordbrood/A = new /mob/living/simple_animal/hostile/asteroid/hivelordbrood(src.loc)
-	A.GiveTarget(target)
-	A.friends = friends
-	A.faction = faction
```
Comment (Firecage) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: Question though. Doesn't this mean, lets say that the Hivelords faction gets changed to syndicate, that the broods wouldn't spawn with the syndicate faction?

File: code/modules/mob/living/simple_animal/hostile/mining_mobs.dm (hunk 1 above)
Comment (Dorsisdwarf) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: Clearly it's talking about your fursuit, cheri :^D

File: code/modules/mob/living/simple_animal/hostile/mining_mobs.dm (hunk 3 above)
Comment (phil235) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: They would spawn with the correct faction if you've edited the hivelord's faction var. The code seems fine to me.

**Comment by phil235** [comment]
The spawn rate is intended, otherwise the hivelord becomes ridiculously easily to kill. It spawns a new broodling everytime it calls its attack target proc, and the broodling only lives for 10 seconds.

I'm closing this, I've borrowed the changes to the hivelord core message so that this part gets merged

**Pull request closed without merging**